src/plugins/dashboard/public/dashboard_app/url/sync_dashboard_url_state.ts

```diff
@@ -117,17 +117,11 @@ export const startSyncingExpandedPanelState = ({
 }) => {
   const expandedPanelSubscription = dashboardAPI?.expandedPanelId.subscribe(() => {
     const expandedPanel = dashboardAPI.expandedPanelId.value;
-    if (expandedPanel) {
-      history.replace({
-        ...history.location,
-        pathname: `${createDashboardEditUrl(dashboardAPI.savedObjectId.value)}/${expandedPanel}`,
-      });
-    } else {
-      history.replace({
-        ...history.location,
-        pathname: createDashboardEditUrl(dashboardAPI.savedObjectId.value),
-      });
-    }
+    const hasExpandedPanel = expandedPanel !== undefined ? `/${expandedPanel}` : '';
+    return history.replace({
+      ...history.location,
+      pathname: `${createDashboardEditUrl(dashboardAPI.savedObjectId.value)}${hasExpandedPanel}`,
+    });
```

Comment:
This looks better than before!! But the name `hasExpandedPanel` makes me expect a `boolean` value, and you are returning a string. I wonder if this is cleaner?

```typescript
history.replace({
   ...history.location,
   pathname: `${createDashboardEditUrl(dashboardAPI.savedObjectId.value)}${Boolean(expandedPanelId) ? `/${expandedPanel}` : ''}`,
}); 
```


Also, `history.replace` doesn't return anything, so why are we returning it?

Reply:
ah cool all in line 👍🏻